src/palo_alto_firewall_analyzer/scripts/pan_analyzer.py

```diff
@@ -119,7 +164,12 @@ def build_output_fname(parsed_args):
     else:
         limit_string = ""
 
-    output_fname = f'pan_analyzer_output_{EXECUTION_START_TIME}{devicegroup_string}{xml_string}{validators_string}{fixers_string}{limit_string}.txt'
+    if parsed_args.output == 'json':
+        extension = '.json'
+    else:    
+        extension = '.txt'
+
+    output_fname = f'pan_analyzer_output_{EXECUTION_START_TIME}{devicegroup_string}{xml_string}{validators_string}{fixers_string}{limit_string}'+extension
```

Comment:
A small nit: Instead of using `+extension`, could you use the existing string formatting and instead add {extension} within the string?

Reply:
fixed